adr/ADR-15.md

```diff
@@ -0,0 +1,233 @@
+# JetStream Subscribe Workflow
+
+|Metadata|Value|
+|--------|-----|
+|Date    |2021-08-11|
+|Author  |@kozlovic|
+|Status  |Partially Implemented|
+|Tags    |jetstream, client|
+
+## Context
+
+This document attempts to describe the workflow of a JetStream subscription in the library, from the creation, the behavior at runtime, to the deletion of a subscription.
+
+## Design
+
+### Creation
+
+The library should have API(s) that allow(s) creation of a JetStream subscription. Depending on the language, there may be various type of subscriptions (like Channel based in Go), or with various options such as Queue subscriptions, etc... There can be different APIs or one with a set of options.
+
+#### Configuration
+
+When creating a subscription, the user can provide several things:
+
+- A subject that is in some cases required (see "Stream name" section below)
+- An optional stream name
+- An optional consumer name
```

Comment:
can this be `An optional durable consumer name`

Reply:
No, it is a consumer name because it does not have to be a durable. For instance, in Go, if you use Bind(stream, consumer) you specify a consumer name, which can be in theory an ephemeral (non durable).